src/err/mod.rs

```diff
@@ -53,6 +53,7 @@ pub struct PyDowncastError<'a> {
 impl<'a> PyDowncastError<'a> {
     /// Create a new `PyDowncastError` representing a failure to convert the object
     /// `from` into the type named in `to`.
+    #[cold]
```

Comment:
This seems unrelated am I am not sure it is a universally good idea. For example, what about `downcast` chains to identify the correct type? We steer users towards downcasting instead of extracting primarily since constructing `PyDowncastError` is cheaper than `PyErr` and this could affect this.

I am not saying this should never happen, just that I would prefer this a separate change with some measurements and discussion on its impact on reasonable code patterns.

Reply:
Agreed this was probably unnecessary, I'll revert and confirm that `codspeed` still thinks the diff is fine. I think the `inline` hints had a much greater effect than this.

I noticed recently that `#[cold]` doesn't prevent inlining and PGO is likely to do better than manual branch hints, so I suspect we should rarely if ever use `#[cold]` to be honest 👍